src/Stratis.Bitcoin.Features.MemoryPool/MempoolManager.cs

```diff
@@ -140,9 +146,17 @@ internal async Task LoadPoolAsync(string fileName = null)
         /// <returns>Memory pool save result.</returns>
         internal MemPoolSaveResult SavePool()
         {
+            this.logger.LogTrace("()");
+
             if (this.mempoolPersistence == null)
+            {
+                this.logger.LogTrace("(-)[NON_SUCCESS]");
                 return MemPoolSaveResult.NonSuccess;
-            return this.mempoolPersistence.Save(this.network, this.memPool);
+            }
+            MemPoolSaveResult saveResult = this.mempoolPersistence.Save(this.network, this.memPool);
+
+            this.logger.LogTrace("(-)[SUCCESS]:{0}.{1}={2},{3}.{4}='{5}'", nameof(MemPoolSaveResult), nameof(saveResult.Succeeded), saveResult.Succeeded, nameof(MemPoolSaveResult), nameof(saveResult.TrxSaved), saveResult.TrxSaved);
```

Comment:
Here I would highly consider having overload for tostring on mempoolsaveresult, which would allow us to do just

```
this.logger.LogTrace("(-):'{0}'", saveResult);
```

Also because the tag is not needed on positive case